File: kafka/src/main/java/com/jtLiBrain/examples/kafka/BrainKafkaZkClient.java
package com.jtLiBrain.examples.kafka;

import kafka.zk.AdminZkClient;
import kafka.zk.KafkaZkClient;
import org.apache.kafka.common.utils.Time;

public class BrainKafkaZkClient {
    private KafkaZkClient kafkaZkClient;
    private AdminZkClient adminZkClient;

    private String zkHost;
    private boolean isSecure = false;
    private int sessionTimeoutMs = 30000;
    private int connectionTimeoutMs = 30000;
    private int  maxInFlightRequests = Integer.MAX_VALUE;
    private Time time = Time.SYSTEM;
    private String metricGroup = "kafka.server";
    private String metricType = "SessionExpireListener";

    public void start() {
        kafkaZkClient = KafkaZkClient.apply(
                zkHost, isSecure, sessionTimeoutMs, connectionTimeoutMs, maxInFlightRequests, time,
                metricGroup, metricType
        );
        adminZkClient = new AdminZkClient(kafkaZkClient);
    }

    public void stop() {

    }
}

File: kafka/src/main/scala/com/jtLiBrain/kafka/BrainKafkaZkClient.scala
package com.jtLiBrain.kafka

import kafka.zk.{AdminZkClient, KafkaZkClient}
import org.apache.kafka.common.utils.Time

class BrainKafkaZkClient {
  private var kafkaZkClient: KafkaZkClient = _
  private var adminZkClient: AdminZkClient = _

  def start(zkHost: String,
            isSecure: Boolean = false,
            sessionTimeoutMs: Int = 30000,
            connectionTimeoutMs: Int = 30000,
            maxInFlightRequests: Int = Int.MaxValue,
            time: Time = Time.SYSTEM,
            metricGroup: String = "kafka.server",
            metricType: String = "SessionExpireListener"
           ): Unit = {
    kafkaZkClient = KafkaZkClient(
      zkHost, isSecure, sessionTimeoutMs, connectionTimeoutMs, maxInFlightRequests, time
    )

    adminZkClient = new AdminZkClient(kafkaZkClient)
  }

  def close(): Unit = {
    if(kafkaZkClient != null) kafkaZkClient.close()
  }
}
